docs/README.md

```diff
@@ -0,0 +1,56 @@
+## Sequence Diagram
+
+This is how the user interacts with the various components of this application.
+
+The different components have been chosen with specific needs in mind:
+* Backend: The backend is needed to protect the private key of the wallet that is doing the actual signing of the transactions on the blockchain.
+* Forwarder: Ensures that the incoming transaction gets validated before actually sent for execution. The forwarder also keeps a record of the internal nonces that external wallets are using while forwarding.
+* Recipient: This is the final smart contract whose method is originally called from the Frontend. It substitutes the forwarder sender for the actual sender (from) of the original transaction and persists the data on-chain.
+
+```mermaid
+  sequenceDiagram
+    participant Frontend
+    participant Metamask
+    participant User
+    participant Backend
+    participant Forwarder
+    participant Recipient
+
+    rect rgb(111, 160, 200, 0.8)
+        Frontend-->>Recipient: Subscribe to Interaction Events
+    end
+
+    activate Metamask
+    rect rgb(208, 240, 192)
+      Frontend->>Metamask: Request Accounts
+      Metamask->>User: Ask to select wallet <br /> and connect
+      User->>Frontend: selects wallet and connects
+    end
+    rect rgb(208, 240, 192)
+      Frontend->>Metamask: Request to INTERACT
+      Metamask->>User: Popup and request to sign a message <br /> (the message contains the final INTERACT step <br/> and the current nonce for the User)
+      User->>Frontend: Signs message
+    end
+    deactivate Metamask
+    activate Backend
+
+    rect rgb(208, 160, 200, 0.9)
+      Frontend->>Backend: Send signed message and sender
```

Comment:
The Frontend sends the signature of the signed message and the request (From (sender Address), To (Recipient Address), Value, Gas, Nonce and Data (a hash of the method that gets called on the recipient SC and the parameter. In this case 'addNewMessage' and Parameter(whatever message the user typed))